processing/src/main/java/io/druid/segment/serde/ComplexColumnPartSerde.java

```diff
@@ -59,21 +58,110 @@ public String getTypeName()
     return typeName;
   }
 
-  @Override
-  public long numBytes()
+  public static SerializerBuilder serializerBuilder()
```

Comment:
I think my confusion mostly arose because ColumnPartSerde now declares new interfaces called Serializer/Deserializer and SerializerBuilder does not build "Serializer" instance but "ComplexMetricSerde" instance. I guess, its a matter of perspective.
you can leave it as is then.